kubetest/e2e.go

```diff
@@ -623,10 +623,35 @@ func kubemarkTest(testArgs []string, dump, numNodes string) error {
 		})
 	}
 
+	if err := control.XmlWrap(&suite, "kubectl version", getKubectlVersion); err != nil {
+		return err
+	}
+
+	if err := control.XmlWrap(&suite, "IsUp", deploy.IsUp); err != nil {
+		return err
+	}
+
 	// Run tests on the kubemark cluster.
 	if err := control.XmlWrap(&suite, "Kubemark Test", func() error {
-		testArgs = util.SetFieldDefault(testArgs, "--ginkgo.focus", "starting\\s30\\pods")
-		return control.FinishRunning(exec.Command("./test/kubemark/run-e2e-tests.sh", testArgs...))
+		testArgs = append(
+			util.SetFieldDefault(testArgs, "--ginkgo.focus", "starting\\s30\\pods"),
+			"--e2e-verify-service-account=false",
+			"--dump-logs-on-failure=false",
+		)
+
+		cmd := exec.Command("./hack/ginkgo-e2e.sh", testArgs...)
+		cmd.Env = append(
+			os.Environ(),
+			"KUBERNETES_PROVIDER=kubemark",
+			"KUBE_CONFIG_FILE=config-default.sh",
+			fmt.Sprintf("KUBECONFIG=%s/kubernetes/test/kubemark/resources/kubeconfig.kubemark", os.Getenv("WORKSPACE")),
+			"E2E_MIN_STARTUP_PODS=0",
+			"KUBE_MASTER_URL=https://"+os.Getenv("KUBE_MASTER_IP"),
```

Comment:
I think this env var is not directly available. We run `detect-master` before this to compute it (see run-e2e-tests.sh).

Reply:
yep, figured, currently got
```
I0314 16:22:24.070] Mar 14 16:22:24.069: INFO: >>> kubeConfig: /workspace/kubernetes/test/kubemark/resources/kubeconfig.kubemark
I0314 16:22:24.072] F0314 16:22:24.071777   10454 e2e.go:154] Error loading client: host must be a URL or a host:port pair: "https://"
```

Reply:
I'll rewrite `detect-master` call as a gcloud call in go here, rather than having all the bash subprocess fun